### ISSUE TITLE
npm audit reports vulnerability with brace package

### ISSUE THREAD

**Issue by lonix1** [jest-community/jest-extended#211]
An `npm audit` fails because this library uses a vulnerable version of the `braces` package.

It uses version 1.8.5, and should be using at least version 2.3.1.

This is a testing tool, so no big deal... but it's really annoying to see the warning every time I use npm.

**Comment by lirantal** [comment]
Agree with @lonix1, would be great to update the package to make sure the vulnerabilities are addressed.

To be more precise however, the vulnerability is in `braces` which is a nested dependency several levels deep as can be seen in the following screenshot from snyk test pages https://snyk.io/test/npm/jest-extended:

![image](https://user-images.githubusercontent.com/316371/55317670-e2358100-5468-11e9-81e5-6335154a5fb1.png)

So the action item to take would be to upgrade the direct dependency of expect to expect@24.0.0 as the Snyk report is proposing, which mitigates the `braces` issue.

**Comment by benjaminkay93** [comment]
@mattphillips just flagging this for you :)

**Comment by lirantal** [comment]
This is still being flagged as a vulnerability until we will upgrade expect on this project.

P.S.
You could get automatic PRs opened with upgrade fix and lockfile re-lock if you connect the repo with snyk. Might help remove this burden.

**Comment by matthew1534** [comment]
There is already a PR for this but it hasn't been merged in: https://github.com/jest-community/jest-extended/pull/198

**Comment by matthew1534** [comment]
This is fixed & merged in, in #198

**Comment by kaidjohnson** [comment]
@matthew1534 Can a new version be cut with this fix so it is available for install via npm/yarn/etc?

**Comment by matthew1534** [comment]
@kaidjohnson I can't do that as I'm not one of the NPM owners, I believe @mattphillips can

**Comment by distante** [comment]
So the NPM package will never be updated?

**Comment by nazrhyn** [comment]
It looks like the last publish for this package was 5 months ago:   
![image](https://user-images.githubusercontent.com/1672405/60834179-0acac680-a18e-11e9-9043-82fd78f3316c.png)

There is a process for "abandoned" packages outlined here: https://www.npmjs.com/policies/disputes   
(See **When to use this process**.) Of course, anyone who started down that path would be taking on the responsibility of maintenance.

<sub>(This is not to say that @mattphillips isn't around, and I mean no disrespect by pointing this out.)</sub>

**Comment by matthew1534** [comment]
@mattphillips could be busy, this package is open source and likely not his full time job... Suggesting trying to force his hand for control of the package is a bit extreme; he's the one that has done us the favour. You can fork/clone the latest version of the repo, which has the fix, and publish it to your own private registry if it is a major issue in your workflows.

This is a testing tool which is in the vast majority of cases is for non-production use so I suspect the Regular Expression Denial of Service vulnerability is not that vital other than it potentially causing `npm audit` to fail if you have it set to not ignore `low` issues.

**Comment by benjaminkay93** [comment]
not tested, but I think this is valid to ignore this if it's just in your `devDepenancies`, I'm not gonna claim its the right thing to do, but until #205 is addressed it might be the best way forward as a tempting solution if its blocking build pipelines.

`npm audit --only=prod`

**Comment by nazrhyn** [comment]
Yeah, I was afraid someone was going to get upset that I even pointed out that process, though I tried to make it clear that wasn't my intention to upset anyone. In response to:

>So the NPM package will never be updated?

The answer is: "Not unless the maintainer does it, or someone else becomes the maintainer." I thought it might be useful to indicate that there is a path forward, in the case that the maintainer does not return.

I agree that the audit issues are not a big deal for development dependencies. That's why I only followed this issue rather than going a different route than using this package.
